crates/cxx-qt-gen/src/parser/qnamespace.rs

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use syn::{ItemMacro, LitStr, Result};
+
+use crate::syntax::attribute::attribute_take_path;
+
+pub struct ParsedQNamespace {
+    /// The name of the namespace
+    pub namespace: String,
+    /// whether qml_element was specified
+    pub qml_element: bool,
+}
+
+impl ParsedQNamespace {
+    pub fn parse(mut mac: ItemMacro) -> Result<Self> {
+        let namespace: LitStr = syn::parse2(mac.mac.tokens)?;
+        let namespace = namespace.value();
```

Comment:
we should check here that the namespace is not empty?

Reply:
That's not done here.
The QEnum parser doesn't care whether the QEnum is assigned to anything, it's still a valid `ParsedQEnum`.

CxxQtData is responsible for matching the QEnum to the corresponding namespace or QObject.
Which will then error out if there's no namespace or QObject to associate the QEnum with.

Reply:
I've added a check that the namespace contains no whitespace and isn't empty.
We may want to do more checks in future, but this should be good enough to cover most cases.